lib/chutney/TorNet.py

```diff
@@ -863,6 +871,15 @@ def getBridge(self):
         except KeyError:
             return 0
 
+    def getEd25519Id(self):
+        """Return the value of ed25519 key"""
+        try:
+            self._setEd25519Id()
```

Comment:
You're calling `_setEd25519Id()` every time you want to get `_env['ed25519_id']`, but you should only need to call it once per node.

The code could work this way, but it's not very efficient.

Can you call `_setEd25519Id()` from `__init__()` instead?

Reply:
hi @teor2345 , can you elaborate this part for me a bit, should I just call the function inside  _init_ ? I am getting errors in doing so.

Reply:
What errors are you getting?

Reply:
FileNotFoundError: [Errno 2] No such file or directory: '/home/anuradha/chutney/net/nodes/002r/keys/ed25519_master_id_public_key'   I did not get the same error while calling the function inside getEd25519Id().

Reply:
That looks like the right path.

Can you push the code you're actually using?
It's hard to give advice when I can't see your code.

Reply:
Where do you get the error in the logs?

Please link to the full logs in CI.

Reply:
https://travis-ci.org/github/torproject/chutney/jobs/675703309 on line 912.